client/src/components/squids/SquidForm.jsx

```diff
@@ -0,0 +1,139 @@
+import React from "react";
+
+import { ErrorMessage } from "@hookform/error-message";
+import { useForm } from "react-hook-form";
+import { useQueryClient } from "react-query";
+
+import { usePostSquid } from "./hooks/usePostSquid";
+
+import "../../style/squids/squidForm.pcss";
+
+export const SquidForm = ({ specialPowers }) => {
+  const {
+    handleSubmit,
+    register,
+    reset: hookReset,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      name: "",
+      species: "",
+      victories: 0,
+      specialPower: undefined,
+    },
+  });
+
+  const { mutate: createSquid } = usePostSquid();
+  const queryClient = useQueryClient();
+
+  const defaultSpecialPowerOption = [
+    <label key="none" className="squid-form__radio-label" htmlFor="none">
+      <input
+        className="squid-form__radio-btn"
+        type="radio"
+        id="none"
+        name="specialPower"
+        value={undefined}
+        {...register("specialPower")}
+      />
+      <span className="squid-form__radio-text">none</span>
+    </label>,
+  ];
+
+  const specialPowerOptions = specialPowers.map((power) => (
```

Comment:
I think there's an opportunity to DRY up the components on lines 29 and 43. What if you create a map like 

```
const specialPowerOptions = [...specialPowers.map(power => ({ id: power, value: power, text: power})), {id: "none", value: undefined, text: "none"}))].map....
```
?

Reply:
YESSSS This is what I wanted to do! Thank you!